bitcoin/src/taproot/merkle_branch.rs

```diff
@@ -252,7 +252,7 @@ impl Iterator for IntoIter {
     fn nth(&mut self, n: usize) -> Option<Self::Item> { self.0.nth(n) }
 
     #[inline]
-    fn last(self) -> Option<Self::Item> { self.0.last() }
+    fn last(mut self) -> Option<Self::Item> { self.0.next_back() }
```

Comment:
I think the linter is wrong here. Since this is a call through and we are implementing trait methods I think we should call through to the same method in the underlying iterator.

Also TIL: I did not know one could add `mut` to the trait method implementation and it still be considered the same method definition.

Reply:
I have changed it to allow the lint.  

I was expecting to have an issue when I added `mut` and was surprised that there wasn't.

Reply:
> I think the linter is wrong here. Since this is a call through and we are implementing trait methods I think we should call through to the same method in the underlying iterator.

I could go either way on this. I think following the linter's suggestion is a bit more future-proof against us changing the type of (`self.0`), and avoids a double-take on the part of the reader.

Reply:
`last` and `next_back` have different semantics. `next_back` is definitely wrong here because they have different side effects. That clippy lint seems sus to me.

Also the `mut` doesn't really change anything. It's equivalent to later rebinding the variable.

Reply:
Of course! Good observation.